test/e2e/metrics_runtime_input_resources_test.go

```diff
@@ -111,19 +127,33 @@ var _ = Describe(suite.ID(), Label(suite.LabelMetrics), Ordered, func() {
 				resp, err := proxyClient.Get(backendOnlyPodMetricsEnabledURL)
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
-				g.Expect(resp).To(HaveHTTPBody(
-					WithFlatMetrics(ContainElement(HaveField("Name", BeElementOf(kubeletstats.PodMetricsNames)))),
-				))
+				bodyContent, err := io.ReadAll(resp.Body)
+				defer resp.Body.Close()
+				g.Expect(err).NotTo(HaveOccurred())
+
+				podMetricNames := slices.Concat(kubeletstats.PodMetricsNames, k8scluster.PodMetricsNames)
+
+				g.Expect(bodyContent).To(WithFlatMetrics(WithNames(
+					OnlyContainElementsOf(podMetricNames))), "Found pod metrics in backend that are not part of k8scluster or kubeletstats")
```

Comment:
also same as above, give it a try to use the built-in `ConsistOf` matcher

Reply:
I had to introduce OnltContainerElementsOf as expected slice does not need to have all the metrics from kubeletstats + k8sCluster. it checks it should not have any other metrics apart from the listed in the expected list. Anf Consists of expects expected slice =result slice (the order can change)